tests/ui/invalid_pymethods.stderr

```diff
@@ -179,11 +179,11 @@ error: macros cannot be used as items in `#[pymethods]` impl blocks
 197 |     macro_invocation!();
     |     ^^^^^^^^^^^^^^^^
 
-error[E0277]: the trait bound `i32: From<&PyType>` is not satisfied
+error[E0277]: the trait bound `i32: From<BoundRef<'_, '_, PyType>>` is not satisfied
```

Comment:
The error message is notably worse than it was before, but I think once we remove the GIL Ref compatibility and go back to just `&Bound<PyType>` it improves. I can live with it being a bit janky for a couple of releases. If reviewers feel strongly that we should try to make this nicer I can experiment at cost of making the implementation more complex.

Reply:
(One upside of a more complex implementation is that I might be able to emit deprecation warnings for using GIL Refs in these positions.)